src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -373,6 +383,9 @@ def _build_storage_profile():
         if disk_info['os'].get('writeAcceleratorEnabled') is not None:
             profile['osDisk']['writeAcceleratorEnabled'] = disk_info['os']['writeAcceleratorEnabled']
         data_disks = [v for k, v in disk_info.items() if k != 'os']
+        for i, data_disk in enumerate(data_disks):
+            if i < len(data_disk_encryption_sets):
```

Comment:
what if data_disks len  not matching with len of data_disk_encryption_sets? better to add check

Reply:
Fixed